# Microbial amplicon sequencing data analysis

This repository provides all data and required tools used in chapter 'Computational and statistical methods to analyse microbial amplicon sequencing data'

## Required tools
1. ##### Amplicon data analysis tools
    * [QIIME1](http://qiime.org/install/install.html) (Quantitative Insights Into Microbial Ecology) - This is open-source OTU based approach for microbial data analysis pipeline
    * [Microbiome helper](https://github.com/LangilleLab/microbiome_helper/wiki) - A repository that contains several resources and scripts to automate microbiome analysis workflows.
    * [DADA2](https://benjjneb.github.io/dada2/index.html) - R package for fast and accurate sample inference from amplicon data with single nucleotide resolution. (ASV based)
2. ##### Quality analysis, filtering, trimming & preprocessing
    * [FASTQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) - A quality control tool for high throughput sequence data.
    * [FASTX-Toolkit](http://hannonlab.cshl.edu/fastx_toolkit/) - FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.
    * [PEAR](https://github.com/tseemann/PEAR) - Paired-End reAd mergeR
3. ##### Other tools
    * [VSEARCH](https://github.com/torognes/vsearch) - an alternative to the USEARCH tool developed by Robert C. Edgar (2010)
4. ##### R packages
    * [phyloseq](https://joey711.github.io/phyloseq/) - The phyloseq package is a tool to import, store, analyze, and graphically display complex phylogenetic sequencing data
    * [ALDEX2](https://bioconductor.org/packages/release/bioc/html/ALDEx2.html) - Analysis Of Differential Abundance Taking Sample Variation Into Account


![]
